frontend/wallet/src/render/views/settings/DelegationSettings.re

```diff
@@ -71,20 +88,54 @@ let make = (~publicKey) => {
   let goBack = () =>
     ReasonReact.Router.push("/settings/" ++ PublicKey.uriEncode(publicKey));
 
-  // Mutation variables and handlers
-  let variables =
-    ChangeDelegation.make(
-      ~from=Apollo.Encoders.publicKey(publicKey),
-      ~to_=Apollo.Encoders.publicKey(PublicKey.ofStringExn(state.delegate)),
-      ~fee=
-        Apollo.Encoders.currency(
-          switch (state.fee) {
-          | DefaultAmount => defaultFee
-          | Custom(amount) => amount
-          },
-        ),
-      (),
-    )##variables;
+  let renderError = () => {
+    switch (state.error) {
+    | Some(EmptyDelegateKey) =>
+      <>
+        <Alert kind=`Danger defaultMessage="Please enter a public key" />
+        <Spacer height=2. />
+      </>
```

Comment:
Since each of these return the same thing maybe we can just switch inside the defaultMessage property of the alert instead. That way we could decrease the duplication in this function.

Reply:
Another option would be to return an optional error message from this function, then use it to conditionally display an alert once

Reply:
Changed it to return an optional string, thanks for the suggestions.